ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/AbstractBlockProcessorIntegrationTest.java

```diff
@@ -0,0 +1,759 @@
+    processContractSlotUpdateThenReadTx(parallelBlockProcessor);
+  }
+
+  @Test
+  void processSequentialSlotReadThenUpdateTx() {
+    processSlotReadThenUpdateTx(sequentialBlockProcessor);
+  }
+
+  @Test
+  void processParallelSlotReadThenUpdateTx() {
+    processSlotReadThenUpdateTx(parallelBlockProcessor);
+  }
+
+  @Test
+  void processSequentialAccountReadThenUpdateTx() {
+    processAccountReadThenUpdateTx(sequentialBlockProcessor);
+  }
+
+  @Test
+  void processParallelAccountReadThenUpdateTx() {
+    processAccountReadThenUpdateTx(parallelBlockProcessor);
+  }
+
+  @Test
+  void processSequentialAccountUpdateThenReadTx() {
+    processAccountUpdateThenReadTx(sequentialBlockProcessor);
+  }
+
+  @Test
+  void processParallelAccountUpdateThenReadTx() {
+    processAccountUpdateThenReadTx(parallelBlockProcessor);
+  }
+
+  @Test
+  void processSequentialAccountReadThenUpdateTxWithTwoAccounts() {
+    processAccountReadThenUpdateTxWithTwoAccounts(sequentialBlockProcessor);
+  }
+
+  @Test
+  void processParallelAccountReadThenUpdateTxWithTwoAccounts() {
+    processAccountReadThenUpdateTxWithTwoAccounts(parallelBlockProcessor);
+  }
+
+  @Test
+  void processSequentialAccountUpdateThenReadTeTxWithTwoAccounts() {
+    processAccountUpdateThenReadTxWithTwoAccounts(sequentialBlockProcessor);
+  }
+
+  @Test
+  void processParallelAccountUpdateThenReadTTxWithTwoAccounts() {
+    processAccountUpdateThenReadTxWithTwoAccounts(parallelBlockProcessor);
+  }
+
+  private void processSimpleTransfers(final BlockProcessor blockProcessor) {
+    // Create two non conflicted transactions
+    Transaction transactionTransfer1 = // ACCOUNT_GENESIS_1 -> ACCOUNT_2
+        createTransferTransaction(
+            0, 1_000_000_000_000_000_000L, 300000L, 5L, 7L, ACCOUNT_2, keyPair1);
+    Transaction transactionTransfer2 = // ACCOUNT_GENESIS_2 -> ACCOUNT_3
+        createTransferTransaction(
+            0, 2_000_000_000_000_000_000L, 300000L, 5L, 7L, ACCOUNT_3, keyPair2);
+
+    MutableWorldState worldState = worldStateArchive.getMutable();
+    BonsaiAccount senderAccount1 = (BonsaiAccount) worldState.get(transactionTransfer1.getSender());
+    BonsaiAccount senderAccount2 = (BonsaiAccount) worldState.get(transactionTransfer2.getSender());
+
+    Block blockWithTransactions =
+        createBlockWithTransactions(
+            "0x4ca6e755674a1df696e5365361a0c352422934ba3ad0a74c9e6b0b56e4f80b4c",
+            transactionTransfer1,
+            transactionTransfer2);
+
+    BlockProcessingResult blockProcessingResult =
+        blockProcessor.processBlock(blockchain, worldState, blockWithTransactions);
+
+    BonsaiAccount updatedSenderAccount1 =
+        (BonsaiAccount) worldState.get(transactionTransfer1.getSender());
+    BonsaiAccount updatedSenderAccount2 =
+        (BonsaiAccount) worldState.get(transactionTransfer2.getSender());
+
+    BonsaiAccount updatedAccount0x1 =
+        (BonsaiAccount) worldState.get(Address.fromHexStringStrict(ACCOUNT_2));
```

Comment:
```suggestion
    BonsaiAccount updatedAccount0x2 =
        (BonsaiAccount) worldState.get(Address.fromHexStringStrict(ACCOUNT_2));
```

Reply:
I'm guessing 0x1 means something else, but I found it confusing to follow the logic because I assumed updatedAccount0x1 referred to ACCOUNT_1 in the assertions further down.

There's a few examples of this naming pattern